src/Features/Core/Portable/InvertConditional/AbstractInvertConditionalCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.InvertConditional
+{
+    internal abstract class AbstractInvertConditionalCodeRefactoringProvider<TConditionalExpressionSyntax> 
+        : CodeRefactoringProvider
+        where TConditionalExpressionSyntax : SyntaxNode
+    {
+        protected abstract bool ShouldOffer(TConditionalExpressionSyntax conditional, int position);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            if (span.Length > 0)
+            {
+                return;
+            }
+
+            var position = span.Start;
+            var conditional = await FindConditionalAsync(document, position, cancellationToken).ConfigureAwait(false);
+
+            if (conditional == null ||
+                position < conditional.Span.Start ||
```

Comment:
>position < conditional.Span.Start [](start = 16, length = 33)

should we also check if position is after the end of the span?

Reply:
sure.